projects/plugins/jetpack/extensions/blocks/map/edit.js

```diff
@@ -221,6 +222,9 @@ class MapEdit extends Component {
 			apiState,
 			apiRequestOutstanding,
 		} = this.state;
+
+		const mapProvider = getMapProvider();
```

Comment:
Personally I would put the assignment outside of the component beside constants definitions. Other than the change is quite straight forward to understand, the changes are neat and when deployed locally it executes as expected. 👍

Reply:
You're right. It shouldn't be called on every render ATM.

However, there's another upcoming ticket for [moving logic to getMapProvider](https://github.com/Automattic/wp-calypso/issues/74048). Implementing this ticket would mean moving it back inside the component (because it would need the `attributes` passed in).

Should I move it out, and move back in later? Or just keep as is?

Reply:
> Should I move it out, and move back in later? Or just keep as is?

In that case let's just leave it in. 😄